src/pudl/extract/excel.py

```diff
@@ -356,3 +365,95 @@ def excel_filename(self, page, **partition):
             string name of the xlsx file
         """
         return self.METADATA.get_file_name(page, **partition)
+
+
+@op
+def merge_yearly_dfs(
```

Comment:
non-blocking: I suppose in theory these don't have to be `_yearly_` - this is generic enough that it should work so long as everything we pass in has the same keys and the DFs that correspond have the same schema. That's pretty nice, we should maybe rename this though - maybe just `merge_dfs_by_page` or something?

Reply:
At some point we'll need to generalize these annual things to work with arbitrary data partitions since some of the other spreadsheets will have other ways of being organized (the PHMSA data has weird irregular multi-year blocks I don't know why) but we haven't done this yet and have been very wedded to yearly chunks for the EIA spreadsheets so far.

Reply:
Great shout, I've renamed as suggested and expanded the docstring to explain what's going on here.